packages/styled-components/src/sheet/Tag.js

```diff
@@ -84,6 +108,29 @@ export class TextTag implements Tag {
     }
   }
 
+  insertRules(startRuleIndex: number, rules: string[]): number {
+    let inserted = 0
+
+    if (startRuleIndex <= this.length && startRuleIndex >= 0) {
+      const fragment = document.createDocumentFragment();
+      const fragmentNodes = fragment.childNodes;
+
+      for (let i = 0, l = rules.length; i < l; i++) {
+        const node = document.createTextNode(rules[i]);
+        const refNode = fragmentNodes[i];
+        fragment.insertBefore(node, refNode || null);
+      }
+
+      const elementRefNode = this.nodes[startRuleIndex]
+      inserted = fragmentNodes.length
+
+      this.element.insertBefore(fragment, elementRefNode)
+      this.length += inserted;
+    }
+
+    return inserted
+  }
```

Comment:
```suggestion
  insertRules(startRuleIndex: number, rules: string[]): number {
    if (startRuleIndex > this.length || startRuleIndex < 0) {
      return 0;
    }
    
    const fragment = document.createDocumentFragment();
    for (let i = 0, l = rules.length; i < l; i++) {
      const node = document.createTextNode(rules[i]);
      fragment.appendChild(node);
    }

    const refNode = this.nodes[startRuleIndex];
    this.element.insertBefore(fragment, refNode || null);
    this.length += rules.length;
    return rules.length;
  }
```

So instead of typing out some random suggestions here, I'm just demonstrating some things in code in form a suggested change 😅 

- `insertBefore` is unnecessary in the fragment loop since we're always appending in order
- `this.length` and the return value can be increased in one go
- add missing `|| null` default to `insertBefore` call at the end

Reply:
thanks! I'd adopted.